src/test/java/rx/observables/SyncOnSubscribeTest.java

```diff
@@ -969,14 +936,73 @@ public void call() {
     }
 
     private void assertJustBehavior(OnSubscribe<FooQux> os) {
-        @SuppressWarnings("unchecked")
-        Observer<Object> o = mock(Observer.class);
+        TestSubscriber<Object> ts = new TestSubscriber<Object>();
+        os.call(ts);
+        ts.assertCompleted();
+        ts.assertNoErrors();
+        ts.assertValueCount(1);
+    }
+    
+    @Test
+    public void testConcurrentUnsubscribe3000Iterations() throws InterruptedException, BrokenBarrierException, ExecutionException{
+        for (int i = 0; i < 3000; i++) {
+            testUnsubscribeConcurrently();
```

Comment:
You could have inlined this loop into the other method because this will create 3000 threads.